aws_lambda_powertools/logging/logger.py

```diff
@@ -605,6 +606,15 @@ def append_context_keys(self, **additional_keys: Any) -> Generator[None, None, N
         with self.registered_formatter.append_context_keys(**additional_keys):
             yield
 
+    def clear_state(self) -> None:
+        """Removes all custom keys that were appended to the Logger."""
+        # Clear all custom keys from the formatter
+        self.registered_formatter.clear_state()
+
+        # Reset to default keys
+        default_keys: dict[Any, Any] = dict(self._default_log_keys)
+        self.structure_logs(**default_keys)
```

Comment:
```suggestion
        self.structure_logs(self._default_log_keys, self.logger_formatter)
```

`self._default_log_keys` is already a dict and immutable for this context, so, we don't to copy this to prevent mutation and stuff. 

The reason I'm adding `self.logger_formatter` is just to keep the API consistent. It shouldn't make any difference in this context, but since `structure_logs` is the function responsible for resolving whether the logger instance will use the custom `logger_formatter` or the default `LambdaPowertoolsFormatter`, then we keep things in order.

Reply:
If I remove `default_keys: dict[Any, Any] = dict(self._default_log_keys)` and keep only the suggested, it gives an error in mypy. 
Keeping as is for now.

Reply:
Actually you're right. Keys in the `structure_logs` are kwargs arguments, good catch!

But you can simple use `self.structure_logs(**self._default_log_keys)` and mypy will be happy.

Reply:
Perfect, this works! Thank you again.